cupy/core/core.pyx

```diff
@@ -2890,6 +2901,80 @@ cpdef ndarray tensordot_core(
     return ret
 
 
+cpdef ndarray tensordot_core_v11(
+        Py_ssize_t transa, Py_ssize_t transb, Py_ssize_t m, Py_ssize_t n,
+        Py_ssize_t k, ndarray a, Py_ssize_t lda, ndarray b, Py_ssize_t ldb,
+        ndarray c, Py_ssize_t ldc):
+    cdef float one_f, zero_f
+    cdef double one_d, zero_d
+    cdef cuComplex one_F, zero_F
+    cdef cuDoubleComplex one_D, zero_D
+
+    if c.dtype.char in 'efF':
+        compute_type = cublas.CUBLAS_COMPUTE_32F
```

Comment:
Don't we wanna use `CUBLAS_COMPUTE_16F` for half precision? Does it not work?

Reply:
You can use `CUBLAS_COMPUTE_16F`, but I didn't use it here for a few reasons.

**Performance:** On a GPU with TensorCore, if the data types of matrix `a`, `b` and `c` are `half` precision, there is little difference in performance of matrix multiply between using `CUBLAS_COMPUTE_16F` and `CUBLAS_COMPUTE_32F` as the compute type.

**Accuracy:** If `CUBLAS_COMPUTE_32F` is used as the compute type, the accumulation in the matrix multiply is performance in `float` precision, which reduces the rounding-error accumulation compared to using `CUBLAS_COMPUTE_16F`, resulting in more accurate results.

**Code maintenance:** If you specify `CUBLAS_COMPUTE_16F` as the compute type, then the parameters `alpha` and `beta` of `cublasGemmEx` must be pointers of `half` dtype. However, `half` is not a 1st citizen in Cython, requiring a bit complicated code. I prefer to keep the source code simple.

Reply:
Thanks, @anaruse! It makes perfect sense 👍